common/net/tlv.py

```diff
@@ -0,0 +1,146 @@
+from common.tools import is_256b_hex_str
+
+import common.net.bigsize as bigsize
+from common.net.utils import message_sanity_checks
+
+tlv_types = {
+    "networks": b"\x01",
+    "amt_to_forward": b"\x02",
+    "outgoing_cltv_value": b"\x04",
+    "short_channel_id": b"\x06",
+    "payment_data": b"\x08",
+}
+
+
+class TLVRecord:
+    """
+    Base class for TLV records.
+
+    Args:
+        t (:obj:`bytes`): the message type.
+        l (:obj:`bytes`): the value length.
+        v (:obj:`bytes`): the message value.
+    """
+
+    def __init__(self, t=b"", l=b"", v=b""):
+        if not isinstance(t, bytes):
+            raise TypeError("t must be bytes")
+        if not isinstance(l, bytes):
+            raise TypeError("l must be bytes")
+        if not isinstance(v, bytes):
+            raise TypeError("v must be bytes")
+
+        self.type = t
+        self.length = l
+        self.value = v
+
+    def __len__(self):
+        """Returns the length of the serialised TLV record"""
+        return len(self.serialize())
+
+    def __eq__(self, other):
+        return isinstance(other, TLVRecord) and self.value == other.value
+
+    @classmethod
+    def from_bytes(cls, message):
+        """
+        Builds a TLV record from bytes.
+
+        Args:
+            message (:obj:`bytes`): the byte representation of the TLV record.
+
+        Returns:
+            :obj:`TLVRecord`: The TLVRecord built from the provided bytes.
+
+        Raises:
+            :obj:`TypeError`: If the provided message is not in bytes.
+            :obj:`ValueError`: If the provided message is not properly encoded.
+        """
+
+        if not isinstance(message, bytes):
+            raise TypeError("message must be bytes")
+
+        try:
+            t, t_offset = bigsize.parse(message)
+            if t.to_bytes(t_offset, "big") == tlv_types["networks"]:
+                return NetworksTLV.from_bytes(message)
+            else:
+                l, l_offset = bigsize.parse(message[t_offset:])
+                v = message[t_offset + l_offset :]
+                if l > len(v):
+                    # Value is not long enough
+                    raise ValueError()  # This message get overwritten so it does not matter
```

Comment:
Might be helpful to define a dict for `t` -> TLV class, since you're likely to add a few going forward :-)